### ISSUE TITLE
Allow dashes and underscores in arg parameter names 

### ISSUE THREAD

**Issue by brikeohane** [palantir/conjure#428]
## Motivation
I'd like to mock an external API found here: https://docs.microsoft.com/en-us/azure/cognitive-services/Translator/reference/v3-0-translate. It would be helpful to define all objects and jersey Service API's within conjure since there is no external SDK available for this service. Then, I could just instantiate the service using JaxRSClient and point it to the correct URI (https://api.cognitive.microsofttranslator.com). This is blocked because the service API contains query parameters that are not within the allowed conjure formatting. In this case, `api-version` contains a dash, which does not match the following regex: `^[a-z]([A-Z]{1,2}[a-z0-9]|[a-z0-9])*[A-Z]?$`. 

## Proposal
Potentially modify this to allow a wider range of parameter names. Is there a reason that we do not allow dashes or underscores in parameter names?

**Comment by uschi2000** [comment]
Did you consider making a standard jax-rs or retrofit API for this service?

**Comment by brikeohane** [comment]
@uschi2000 Yes, that is the alternative approach that I am taking. Barely any more lift than using Conjure, so this isn't blocking; more just curiosity
